### DISCUSSION TITLE
Request for help on measuring the dimensions of an object from Roboflow data.

### DISCUSSION THREAD

**Discussion by HopeUptone** [ultralytics/ultralytics#14237] [Q&A]
Good morning
I am working on a project to determine the dimensions of crushed gravel. I first proceeded to annotate the different diameters of the gravel using the segmentation instance in roboflow. I am now finished. I would like to import the dataset in the form of yolov8 code and train my model on google colab.
Now I don't know how to determine the dimensions of the gravel found in the bounding box. Help me please 
![Uploading cap roboflow.png…]()

**Comment by glenn-jocher** [comment]
@HopeUptone good morning!

Thank you for reaching out. To measure the dimensions of objects like crushed gravel using YOLOv8, you can follow these steps:

1. **Import and Train Your Dataset**: First, ensure your dataset is correctly formatted for YOLOv8. You can use the `roboflow` library to download and convert your dataset to YOLOv8 format. Here's a basic example:

    ```python
    from roboflow import Roboflow

    rf = Roboflow(api_key="YOUR_API_KEY")
    project = rf.workspace().project("YOUR_PROJECT_NAME")
    dataset = project.version("YOUR_VERSION_NUMBER").download("yolov8")
    ```

2. **Train Your Model**: Use the Ultralytics YOLOv8 library to train your model on Google Colab:

    ```python
    from ultralytics import YOLO

    # Load a model
    model = YOLO('yolov8n.yaml')  # build a new model from YAML
    model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

    # Train the model
    model.train(data='path/to/your/dataset.yaml', epochs=100, imgsz=640)
    ```

3. **Measure Object Dimensions**: After training, you can use the model to predict and measure the dimensions of the objects. Here's an example of how to do this:

    ```python
    results = model.predict(source='path/to/your/image.jpg')
    for result in results:
        for box in result.boxes:
            x1, y1, x2, y2 = box.xyxy[0]  # Extract bounding box coordinates
            width = x2 - x1
            height = y2 - y1
            print(f"Width: {width}, Height: {height}")
    ```

For more detailed guidance, please refer to our [documentation](https://docs.ultralytics.com). If you encounter any issues, providing a minimum reproducible example will help us assist you better. You can find more information on creating one [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Lastly, ensure you are using the latest versions of the packages to avoid any compatibility issues.

Happy coding! 😊

**Comment by HopeUptone** [comment]
Good morning, Mr. Glenn Jocher. Thanks. I'd like to ask if the script to
have the dimensions of the objects is to be run in Google colab.

Le sam. 6 juil. 2024 à 11:37, Glenn Jocher ***@***.***> a
écrit :

**Reply by pderrenger** [reply]
@HopeUptone good morning,

Thank you for your question! To measure the dimensions of objects like crushed gravel using YOLOv8, you can indeed run the script in Google Colab. Here's a step-by-step guide to help you get started:

1. **Import and Train Your Dataset**: First, ensure your dataset is correctly formatted for YOLOv8. You can use the Roboflow library to download and convert your dataset to YOLOv8 format. Here's a basic example:

    ```python
    from roboflow import Roboflow

    rf = Roboflow(api_key="YOUR_API_KEY")
    project = rf.workspace().project("YOUR_PROJECT_NAME")
    dataset = project.version("YOUR_VERSION_NUMBER").download("yolov8")
    ```

2. **Train Your Model**: Use the Ultralytics YOLOv8 library to train your model on Google Colab:

    ```python
    from ultralytics import YOLO

    # Load a model
    model = YOLO('yolov8n.yaml')  # build a new model from YAML
    model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

    # Train the model
    model.train(data='path/to/your/dataset.yaml', epochs=100, imgsz=640)
    ```

3. **Measure Object Dimensions**: After training, you can use the model to predict and measure the dimensions of the objects. Here's an example of how to do this:

    ```python
    results = model.predict(source='path/to/your/image.jpg')
    for result in results:
        for box in result.boxes:
            x1, y1, x2, y2 = box.xyxy[0]  # Extract bounding box coordinates
            width = x2 - x1
            height = y2 - y1
            print(f"Width: {width}, Height: {height}")
    ```

For more detailed guidance, please refer to our [documentation](https://docs.ultralytics.com). If you encounter any issues, providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example) will help us assist you better.

Lastly, ensure you are using the latest versions of the packages to avoid any compatibility issues.

Happy coding! 😊

Feel free to reach out if you have any more questions.
